src/NServiceBus.Core/Pipeline/Incoming/InvokeHandlerTerminator.cs

```diff
@@ -49,6 +51,16 @@ await messageHandler
 
                 activity?.SetErrorStatus(ex);
 
+                //context.Extensions.SetOnRoot(ex.GetHashCode(), messageHandler.HandlerType);
+
+                if (messageHandler.Instance.GetType().GetInterface("IUseRecoverabilityConfiguration`1") is { } t)
+                {
+                    var configType = t.GetGenericArguments().Single();
+                    var config = (IRecoverabilityConfiguration)Activator.CreateInstance(configType);
+                    var recoverabilityConfiguration = config.OnError(ex, context.MessageBeingHandled, context.Headers, context.Extensions);
```

Comment:
Is there no way to expose the LogicalMessage here instead of object like we do on the errorcontext?

Reply:
sure we could also expose the LogicalMessage, but would that bring any benefits?